src/mbgl/text/placement.cpp

```diff
@@ -661,12 +656,8 @@ void Placement::placeBucket(const SymbolBucket& bucket,
         }
 
     } else {
-        auto beginIt = bucket.symbolInstances.begin() + params.symbolInstanceStart;
-        auto endIt = bucket.symbolInstances.begin() + params.symbolInstanceEnd;
-        assert(params.symbolInstanceStart < params.symbolInstanceEnd);
-        assert(params.symbolInstanceEnd <= bucket.symbolInstances.size());
-        for (auto it = beginIt; it != endIt; ++it) {
-            placeSymbol(*it);
+        for (const SymbolInstance& symbol : bucket.getSymbols(params.sortKeyRange)) {
```

Comment:
before, symbol instances were iterated in-place, now getSymbols creates new vector for the same operation.